source/client/factories_impl.cc

```diff
@@ -31,18 +31,28 @@ BenchmarkClientPtr BenchmarkClientFactoryImpl::create(
     Envoy::Api::Api& api, Envoy::Event::Dispatcher& dispatcher, Envoy::Stats::Scope& scope,
     Envoy::Upstream::ClusterManagerPtr& cluster_manager,
     Envoy::Tracing::HttpTracerSharedPtr& http_tracer, absl::string_view cluster_name,
-    RequestSource& request_generator) const {
+    const int sink_stat_prefix, RequestSource& request_generator) const {
   StatisticFactoryImpl statistic_factory(options_);
   // While we lack options to configure which statistic backend goes where, we directly pass
   // StreamingStatistic for the stats that track response sizes. Ideally we would have options
   // for this to route the right stat to the right backend (HdrStatistic, SimpleStatistic,
   // NullStatistic).
   // TODO(#292): Create options and have the StatisticFactory consider those when instantiating
   // statistics.
+  BenchmarkClientStatistic statistic;
```

Comment:
Would it make sense to define a constructor for this struct right next to it and just call the constructor here?

Reply:
Thanks for the suggestion. Added a constructor for the struct. Yes, this makes the code simpler.